fedimint-api-client/src/query.rs

```diff
@@ -388,302 +385,3 @@ impl<R> QueryStrategy<R, BTreeMap<PeerId, R>> for ThresholdOrDeadline<R> {
         }
     }
 }
-
-/// Query for supported api versions from all the guardians (with a deadline)
-/// and calculate the best versions to use for each component (core + modules).
-pub struct DiscoverApiVersionSet {
-    inner: ThresholdOrDeadline<SupportedApiVersionsSummary>,
-    client_versions: SupportedApiVersionsSummary,
-}
-
-impl DiscoverApiVersionSet {
-    pub fn new(
-        threshold: usize,
-        deadline: SystemTime,
-        client_versions: SupportedApiVersionsSummary,
-    ) -> Self {
-        Self {
-            inner: ThresholdOrDeadline::new(threshold, deadline),
-            client_versions,
-        }
-    }
-}
-
-impl QueryStrategy<SupportedApiVersionsSummary, ApiVersionSet> for DiscoverApiVersionSet {
-    fn request_timeout(&self) -> Option<Duration> {
-        Some(
-            self.inner
-                .deadline
-                .duration_since(fedimint_core::time::now())
-                .unwrap_or(Duration::ZERO),
-        )
-    }
-
-    fn process(
-        &mut self,
-        peer: PeerId,
-        result: api::PeerResult<SupportedApiVersionsSummary>,
-    ) -> QueryStep<ApiVersionSet> {
-        match self.inner.process(peer, result) {
-            QueryStep::Success(o) => {
-                match discover_common_api_versions_set(&self.client_versions, o) {
-                    Ok(o) => QueryStep::Success(o),
-                    Err(e) => QueryStep::Failure {
-                        general: Some(e),
-                        peers: BTreeMap::new(),
-                    },
-                }
-            }
-            QueryStep::Retry(v) => QueryStep::Retry(v),
-            QueryStep::Continue => QueryStep::Continue,
-            QueryStep::Failure { general, peers } => QueryStep::Failure { general, peers },
-        }
-    }
-}
-
-fn discover_common_core_api_version(
```

Comment:
This logic + tests for it were moved to a new file unmodified. They possibly suck and could use some thinking, but it has basic tests, so it shouldn't explode anytime soon.